consensus/propagation/requester.go

```diff
@@ -0,0 +1,252 @@
+package propagation
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	proptypes "github.com/tendermint/tendermint/consensus/propagation/types"
+	"github.com/tendermint/tendermint/libs/bits"
+	"github.com/tendermint/tendermint/libs/log"
+	"github.com/tendermint/tendermint/libs/rand"
+	"github.com/tendermint/tendermint/p2p"
+	"github.com/tendermint/tendermint/proto/tendermint/propagation"
+)
+
+const (
+	// concurrentPerPeerRequestLimit the maximum number of requests to send a peer.
+	concurrentPerPeerRequestLimit = 100
+
+	// maxRequestsPerPart the maximum number of requests per parts.
+	maxRequestsPerPart = 30
+
+	// maxNumberOfPendingRequests the maximum number of pending requests.
+	maxNumberOfPendingRequests = 50_000
+
+	// maxRequestRetry the maximum number of times to try to send a request
+	maxRequestRetry = 10
+
+	// requestTimeout request timeout after it's pending to be sent
+	requestTimeout = 60 * time.Second
```

Comment:
@evan-forbes note: updating these params to lower values make the e2e test fail. I put these values for now because they make the e2e pass. However, we should give them more thoughts when moving this to production.

This makes sense because the e2e tests malicious scenarios. So, if we set, for example, max requests per part to 2, if those requests land on malicious nodes, we will not receive the data. 

Also, should I create an issue for this?